mixer/pkg/runtime/config/snapshot.go

```diff
@@ -61,6 +81,16 @@ type (
 		Params proto.Message
 	}
 
+	// Instance2 configuration. Fully resolved.
+	Instance2 struct {
+		Name string
+
+		Template *Template
+
+		// Encoder to create request instance bytes from attributes
+		Encoder *dynamic.Encoder
```

Comment:
`Encoder *dynamic.Encoder` is not part of the configuration.
instance2 should have name, template and input to the EncoderBuilder.
We should create the Encoder as part of creating the runtime object.

Reply:
For the purpose of validating the user written instance, we will be creating the `Encoder` during snapshot building phase. IMO encoder is not super different that the user config input, it is just a different view of it, why should we not just cache it inside snapshot itself ? WDYT ?

Reply:
what is the equivalent field on the generated code side?

Reply:
There is a builder that we compute during a routing table building phase..
https://github.com/istio/istio/blob/66434029ad81f1d6e049c413c5150908d147d9d1/mixer/pkg/runtime/routing/builder.go#L223

We should do this `builder` creation during snapshot building phase. During routing table building we should just connect to the handler, build it and reference the pre built artifacts inside the routing table entries. Done.

Reply:
He is storing the result of validation which is the encoder and an error.
Validation can throw away the result and call build again at a later time.

```
encoder, err := builder.build(protodecs, bytes)
```